### ISSUE TITLE
[$250] Chat - Double strike through is applied for emojis

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#54618]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.79-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**Issue reported by:** Applause Internal Team
**Device used:** Redmi note 10s Android 13
**App Component:** Chat Report View

## Action Performed:
1. Go to https://staging.new.expensify.com/home
2. Open a chat 
3. Enter `~😗~`
4. Send the message

## Expected Result:
Double strike through must not be applied for emojis.

## Actual Result:
Double strike through is applied for emojis.

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/0ba6e3d8-2f02-4162-927f-8b2970b1c033

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021876648717605677908</li>
        <li>Upwork Job ID: 1876648717605677908</li>
        <li>Last Price Increase: 2025-01-28</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

**Comment by Shahidullah-Muffakir** [comment]
It will be fixed by this PR https://github.com/Expensify/react-native-live-markdown/pull/591

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Double strike-through is seen on emoji.

### What is the root cause of that problem?
We previously prevented the emoji from being formatted in https://github.com/Expensify/App/pull/40617. The strikethrough style is applied from the parent (`<del>`) and the emoji itself. In that PR, we apply an inline-flex style so the emoji won't be affected by the parent strikethrough style.

But in https://github.com/Expensify/App/pull/47547, we changed it to inline because selecting the text with emoji won't have the same height.
<img width="101" alt="image" src="https://github.com/user-attachments/assets/95ebdf10-3644-4646-839d-7d70cf4c24f3" />

Compared to when using inline.
<img width="93" alt="image" src="https://github.com/user-attachments/assets/65f6df99-3420-48c1-95bb-4637ae5732e1" />


### What changes do you think we should make in order to solve the problem?

I think it's more important to prevent the emoji from being formatted compared to making sure the selection height to be same and it's barely noticeable too, so I suggest to use inline-flex back and add `textDecorationLine: 'none' to override the text decoration style on the emoji itself.
https://github.com/Expensify/App/blob/e94df0f79443f99b33be867fec8637c6a9aa5821/src/styles/utils/emojiDefaultStyles/index.ts#L6-L9

```
const emojiDefaultStyles: EmojiDefaultStyles = {
    ...FontUtils.fontFamily.platform.EXP_NEUE,
    ...display.dInlineFlex,
    textDecorationLine: 'none',
};
```

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?

N/A

**Comment by bernhardoj** [comment]
It was solved months ago, but it happens again.

**Comment by bfitzexpensify** [comment]
I agree that this is a duplicate of https://github.com/Expensify/App/issues/14676

**Comment by bernhardoj** [comment]
@bfitzexpensify I wouldn't say it's a dupe because that issue is solved months ago, but this issue happens again, so we should fix it again.

**Comment by bernhardoj** [comment]
It's not a live markdown issue.

**Comment by bernhardoj** [comment]
I have no idea for that.

**Comment by bfitzexpensify** [comment]
No need for a bounty increase, this is a low priority issue.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by bfitzexpensify** [comment]
Ah, so the 'not reproducible' was basically saying we don't apply a double-strikethrough visually, but we are still applying a strikethrough when we shouldn't (and even though it looks like a single strikethrough, it's a double, just on the same line). 

I agree with you. Reopening.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by bfitzexpensify** [comment]
No update here

**Comment by bfitzexpensify** [comment]
I'm going to close this instead. I don't really see a major issue with this behaviour, especially since it only occurs after intentionally trying to create a strikethrough on the emoji.
